blockware/lib/VirtualBlox/stubs/LIS2DW12Sensor.h

```diff
@@ -0,0 +1,262 @@
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ ******************************************************************************
+ */
+
+
+/* Prevent recursive inclusion -----------------------------------------------*/
+
+#ifndef __LIS2DW12Sensor_H__
+#define __LIS2DW12Sensor_H__
+
+
+/* Includes ------------------------------------------------------------------*/
+
+#include "Wire.h"
+// #include "SPI.h"
+// #include "lis2dw12_reg.h"
+
+/* Defines -------------------------------------------------------------------*/
+
+#define LIS2DW12_ACC_SENSITIVITY_FOR_FS_2G_LOPOW1_MODE   0.976f  /**< Sensitivity value for 2g full scale, Low-power1 mode [mg/LSB] */
+#define LIS2DW12_ACC_SENSITIVITY_FOR_FS_2G_OTHER_MODES   0.244f  /**< Sensitivity value for 2g full scale, all other modes except Low-power1 [mg/LSB] */
+
+#define LIS2DW12_ACC_SENSITIVITY_FOR_FS_4G_LOPOW1_MODE   1.952f  /**< Sensitivity value for 4g full scale, Low-power1 mode [mg/LSB] */
+#define LIS2DW12_ACC_SENSITIVITY_FOR_FS_4G_OTHER_MODES   0.488f  /**< Sensitivity value for 4g full scale, all other modes except Low-power1 [mg/LSB] */
+
+#define LIS2DW12_ACC_SENSITIVITY_FOR_FS_8G_LOPOW1_MODE   3.904f  /**< Sensitivity value for 8g full scale, Low-power1 mode [mg/LSB] */
+#define LIS2DW12_ACC_SENSITIVITY_FOR_FS_8G_OTHER_MODES   0.976f  /**< Sensitivity value for 8g full scale, all other modes except Low-power1 [mg/LSB] */
+
+#define LIS2DW12_ACC_SENSITIVITY_FOR_FS_16G_LOPOW1_MODE  7.808f  /**< Sensitivity value for 16g full scale, Low-power1 mode [mg/LSB] */
+#define LIS2DW12_ACC_SENSITIVITY_FOR_FS_16G_OTHER_MODES  1.952f  /**< Sensitivity value for 16g full scale, all other modes except Low-power1 [mg/LSB] */
+
+/* Typedefs ------------------------------------------------------------------*/
+typedef enum
+{
+  LIS2DW12_STATUS_OK = 0,
+  LIS2DW12_STATUS_ERROR
+} LIS2DW12StatusTypeDef;
+
+typedef struct
+{
+  unsigned int WakeUpStatus : 1;
+  unsigned int D6DOrientationStatus : 1;
+  unsigned int SleepStatus : 1;
+} LIS2DW12_Event_Status_t;
+
+typedef enum
+{
+  LIS2DW12_HIGH_PERFORMANCE_MODE,
+  LIS2DW12_LOW_POWER_MODE4,
+  LIS2DW12_LOW_POWER_MODE3,
+  LIS2DW12_LOW_POWER_MODE2,
+  LIS2DW12_LOW_POWER_MODE1
+} LIS2DW12_Operating_Mode_t;
+
+typedef enum
+{
+  LIS2DW12_LOW_NOISE_DISABLE,
+  LIS2DW12_LOW_NOISE_ENABLE
+} LIS2DW12_Low_Noise_t;
+
+/* Class Declaration ---------------------------------------------------------*/
+   
+/**
+ * Abstract class of an LIS2DW12 Inertial Measurement Unit (IMU) 3 axes
+ * sensor.
+ */
+class LIS2DW12Sensor
+{
+  public:
+    LIS2DW12Sensor(TwoWire *i2c, uint8_t address=0);
+    // LIS2DW12Sensor(SPIClass *spi, int cs_pin, uint32_t spi_speed=2000000);
```

Comment:
My workflow was just copy in the original files, comment everything out, then selectively add back whatever was needed to compile.